openedx_proversity_reports/utils.py

```diff
@@ -287,3 +287,78 @@ def get_attribute_from_module(module, attribute_name):
     """
     module = import_module(module)
     return getattr(module, attribute_name, None)
+
+
+def get_exisiting_users_by_email(user_email_list):
+    """
+    Return a list of django.contrib.auth.models.User instances of users
+    that exists in the platform.
+
+    Args:
+        user_email_list: List containing the emails of the users.
+    Returns:
+        exisiting_user_list: List containing django.contrib.auth.models.User instances.
+    """
+    exisiting_user_list = []
+
+    for user_email in user_email_list:
+        try:
+            user = User.objects.get(email=user_email)
+            exisiting_user_list.append(user)
+        except User.DoesNotExist:
+            continue
+
+    return exisiting_user_list
+
+
+def get_user_course_enrollments(user):
+    """
+    Return a list of course keys of the courses where the user is enrolled.
+
+    Args:
+        user: django.contrib.auth.models.User instance.
+    Returns:
+        List containing opaque_keys.edx.keys.CourseKey course instances.
+    """
+    course_enrollment_objects = get_course_enrollment().objects
+    user_course_enrollments = course_enrollment_objects.filter(
+        user__id=user.id,
+        user__courseaccessrole__id=None,
```

Comment:
why is this necessary ?

Reply:
To get the course enrollments only for students.